src/types.ts

```diff
@@ -114,3 +114,29 @@ export type VideoAtomType = {
     height?: number;
     width?: number;
 };
+
+type Weighting =
+    | 'inline'
+    | 'thumbnail'
+    | 'supporting'
+    | 'showcase'
+    | 'halfwidth'
+    | 'immersive';
+
+export type SrcSetItem = { src: string; width: number };
+
+export type ImageSource = {
+    weighting: Weighting;
+    srcSet: SrcSetItem[];
+};
+
+// aka weighting. RoleType affects how an image is placed. It is called weighting
+// in Composer but role in CAPI. We respect CAPI so we maintain this nomenclature
+// in DCR
+export type RoleType =
+    | 'immersive'
+    | 'supporting'
+    | 'showcase'
+    | 'inline'
+    | 'thumbnail'
+    | 'halfWidth';
```

Comment:
I can duplicate the types in `Youtube` and in `Picture` I need them for when you pass down props

Reply:
Ah, I wasn't sure if they were used in multiple places or not.

Aside: Do we still need both weighting and role? If I remember right, this was an issue in DCR because Richlinks use a different type.

Reply:
🤷  No idea of its origin. I guess we can remove it then? Im a little worried as we would start to diverge from DCR then